src/ol/interaction/PolygonBrush.js

```diff
@@ -0,0 +1,146 @@
+import {polygon as turfPolygon} from '@turf/helpers'
+import booleanOverlap from '@turf/boolean-overlap'
+import Draw from './Draw.js';
+import {DrawEvent} from './Draw.js';
+import {DrawEventType} from './Draw.js';
+import Circle from '../geom/Circle.js';
+import Feature from '../Feature.js';
+import Polygon from '../geom/Polygon.js'
+import MapBrowserEventType from '../MapBrowserEventType.js';
+import EventType from '../events/EventType.js';
+import {shiftKeyOnly} from '../events/condition.js';
+import {fromCircle} from '../geom/Polygon.js';
+import {union} from '../geom/flat/union.js';
+
+class PolygonBrush extends Draw {
+
+  constructor(options) {
+
+    super(options)
+    this.circleRadius_ = 10000  //TODO better value
+    this.drawmode_ = false;
+    this.newFeature = null;
+  }
+
+  handleEvent(event) {
+    let pass = super.handleEvent(event);
+    const type = event.type;
+    const btn = event.originalEvent.button;
+    if (shiftKeyOnly(event) && (type === EventType.WHEEL || type === EventType.MOUSEWHEEL)) {
+      pass = false;
+      this.updateSketchPointRadius_(event);
+    }
+    if (btn == 0 && (type === MapBrowserEventType.POINTERDOWN)) {
+      pass = false;
+      this.drawmode_ = true;
+      this.startDrawing_(event);
+      this.continueDrawing();
+      this.createOrUpdateSketchPoint_(event);
+    }
+    if (this.drawmode_ && type === MapBrowserEventType.POINTERMOVE) {
+      pass = false;
+      this.startDrawing_(event);
+      this.continueDrawing();
+      this.createOrUpdateSketchPoint_(event);
+    }
+    if (btn == 0 && this.drawmode_ && type === MapBrowserEventType.POINTERUP) {
+      this.finishDrawing();
+//      this.createOrUpdateSketchPoint_(event);
+    }
+    return pass
+  }
+
+  handleUpEvent(event) {
+    //this is overridden because nothing should happen here
```

Comment:
Sollte hier nicht das `drawend` Event gefeuert werden? Oder wird das im Original Code auch alles in `handleEvent` gemacht?

Reply:
Habe das Event jetzt entsprechend hinzugefügt.